src/domain/pricing/bulk-editor/index.tsx

```diff
@@ -0,0 +1,1052 @@
+  const [currentEditAmount, setCurrentEditAmount] = useState<
+    string | undefined
+  >()
+
+  const { current: pointers } = useRef<CellPointers>({} as CellPointers)
+
+  /**
+   * A struct for mapping between table cell indexes and variant/regions ids.
+   */
+  const matrix = useMemo(() => {
+    const variants = products.reduce(
+      (acc, p) => [...acc, ...p.variants.map((v) => v.id)],
+      []
+    )
+
+    return [variants, activeRegions.map((r) => r.id)]
+  }, [activeRegions, products])
+
+  useEffect(() => {
+    const handler = (e) => {
+      const isPriceInputClicked = e.target.classList.contains("js-bt-input")
+
+      // artificial blur event
+      if (!isPriceInputClicked) {
+        setCurrentEditAmount(undefined)
+        setActiveFields({})
+      }
+    }
+
+    document.addEventListener("mousedown", handler)
+    return () => document.removeEventListener("mousedown", handler)
+  }, [])
+
+  /* ********** HELPERS ********** */
+
+  const activateField = (cellKey: string) => {
+    setActiveFields({ ...activeFields, [cellKey]: true })
+  }
+
+  const deactivateField = (cellKey: string) => {
+    const tmp = { ...activeFields }
+    delete tmp[cellKey]
+
+    setActiveFields(tmp)
+  }
+
+  const toggleActive = (cellKey: string) => {
+    if (activeFields[cellKey]) {
+      deactivateField(cellKey)
+    } else {
+      activateField(cellKey)
+    }
+  }
+
+  const setActiveCellsFromPointers = () => {
+    const next = {}
+    const [variants, regions] = matrix
+
+    for (let i = pointers.minV; i <= pointers.maxV; i++) {
+      const v = variants[i]
+      for (let j = pointers.minR; j <= pointers.maxR; j++) {
+        const r = regions[j]
+        next[getPriceKey(v, r)] = true
+      }
+    }
+
+    setActiveFields(next)
+  }
+
+  /* ********** SELECTORS ********** */
+
+  const isActive = (variantId: string, regionId: string) =>
+    activeFields[getPriceKey(variantId, regionId)]
+
+  const getPriceChange = (variantId: string, regionId: string) => {
+    const key = getPriceKey(variantId, regionId)
+
+    if (key in priceChanges && !priceChanges[key]) {
+      return null
+    }
+    return priceChanges[key]
+  }
+
+  /* ********** HANDLERS ********** */
+
+  /*
+   * A handler that implement logic for mouse click multi select.
+   * Responsible for triggering focus/blur events on an input filed.
+   */
+  const onPriceInputClick = (
+    e: React.MouseEvent<HTMLInputElement>,
+    variantId: string,
+    regionId: string
+  ) => {
+    const cellKey = getPriceKey(variantId, regionId)
+
+    if (e.shiftKey) {
+      e.preventDefault() // do not focus
+
+      const active = isActive(variantId, regionId)
+      toggleActive(cellKey)
+
+      if (active) {
+        const wasFocused = e.target === document.activeElement
+        e.target.blur()
+
+        if (wasFocused) {
+          // find another active cell to focus
+          const id = Object.keys(activeFields).find((k) => k !== cellKey)
```

Comment:
Tell me if I am wrong but here we could focus any active field before or after the one that was currently active? since the activeFields does not hold any order, could that lead to weird behaviour from the end user point of view?

Reply:
At this point, a user is editing multiple cells which have a "virtual" focus so it doesn't matter much from the user's point of view which one is actually focused in the DOM.